lib/commands/asm/network/virtualNetwork._js

```
@@ -0,0 +1,690 @@
+          output.warn(util.format($('A virtual network with name "%s" not found'), vnetName));
+        }
+      }
+    } else {
+      if (output.format().json) {
+        output.json({});
+      } else {
+        output.warn(util.format($('A virtual network with name "%s" not found'), vnetName));
+      }
+    }
+  },
+
+  delete: function (vnetName, options, _) {
+    var networkConfig = this.networkConfig.get(_);
+    var vnetConfig = networkConfig.VirtualNetworkConfiguration;
+    var output = this.cli.output;
+    var interaction = this.cli.interaction;
+
+    var index = utils.indexOfCaseIgnore(vnetConfig.VirtualNetworkSites, {Name: vnetName});
+    if (index) {
+      if (!options.quiet && !interaction.confirm(util.format($('Delete a virtual network "%s"? [y/n] '), vnetName), _)) {
+        return;
+      }
+      vnetConfig.VirtualNetworkSites.splice(index, 1);
+      this.networkConfig.set(networkConfig, _);
+    } else {
+      output.error(util.format($('A virtual network with name "%s" not found'), vnetName));
+    }
+  },
+
+  _getSites: function (_) {
+    var progress = this.cli.interaction.progress($('Looking up the virtual network sites'));
+    try {
+      var response = this.networkManagementClient.networks.list(_);
+      return response.virtualNetworkSites;
+    } catch (e) {
+      return null;
+    } finally {
+      progress.end();
+    }
+  },
+
+  _getLocations: function (_) {
+    var progress = this.cli.interaction.progress($('Looking up locations'));
+    try {
+      var response = this.managementClient.locations.list(_);
+      return response.locations;
+    } catch (e) {
+      return null;
+    } finally {
+      progress.end();
+    }
+  },
+
+  _getAffinityGroups: function (_) {
+    var progress = this.cli.interaction.progress($('Looking up affinity groups'));
+    try {
+      var response = this.managementClient.affinityGroups.list(_);
+      return response.affinityGroups;
+    } catch (e) {
+      return null;
```

why swallow the error?